### DISCUSSION TITLE
Questions on attributes

### DISCUSSION THREAD

**Discussion by Aya-Badawy89** [gama-platform/gama#99] [Q&A] [answered]
Suppose we have this code:
```
species aya skills:[fipa] {
	int x <- 4;
	int y <- 5;
	
	reflex {
		write species(self).attributes;
	}
}
```
How can I make the agent get the value of the attribute whose name is (x)? 
I know of course that Gama does that automatically if I just write (x). However, I wanted the agent to get the value using the string name of the variable because it can help in communicating and argumenting. For example, when an agent tells another agent an information about the variable whose name is (x), the agent should be able to go to this variable by its name and check it, then they can talk about it. So, is it possible to get values of variables with their string names? 
 
Thank you in advance

**Comment by benoitgaudou** [comment, marked as answer]
Dear,
I think that  what you are looking for is the *get* operator.

see below an example:

*model* NewModel


*global* {

*init* {

*create* *people*;

*ask* *people* {

*write* *species*(*self*).*attributes*;

*loop* n  *over*: *species*(*self*).*attributes* {

*write self get(n);*

}

}

}

}


*species* people *skills*: [*moving*] {}

Cheers

Benoit

Le mar. 7 sept. 2021 à 20:16, Aya-Badawy89 ***@***.***> a
écrit :

**Reply by Aya-Badawy89** [reply]
Thank you very much. I did not know about it.
